.github/workflows/ci.yaml

```diff
@@ -39,25 +38,55 @@ jobs:
       - name: Clone OpenAssetIO
         uses: actions/checkout@v2
         with:
-          repository: TheFoundryVisionmongers/OpenAssetIO
+          repository: foundrytom/OpenAssetIO
+          ref: work/347-pythonVenvBuild
           path: dependencies/OpenAssetIO
 
+      - name: Cache conan packages
+        id: cache-conan
+        uses: actions/cache@v2
+        with:
+          path: ~/conan
+          key: ubuntu-latest-conan
+
+      - name: Build OpenAssetIO
+        run: |
+          pip install conan==1.45.0 cmake==3.21 ninja==1.10.2.3 cpplint==1.5.5
+          conan profile new default --detect --force
+          conan profile update settings.compiler.libcxx=libstdc++ default
+          conan install --install-folder ~/conan --build=missing dependencies/OpenAssetIO/resources/build
+          cmake -S dependencies/OpenAssetIO -B build  \
+            --toolchain ~/conan/conan_paths.cmake \
+            --install-prefix $GITHUB_WORKSPACE/dist
+          cmake --build build --target openassetio-python-py-install
+
+      - name: Clone OpenAssetIO-MediaCreation
+        uses: actions/checkout@v2
+        with:
+          repository: foundrytom/OpenAssetIO-MediaCreation
+          ref: work/347-timelineTraits
+          path: dependencies/OpenAssetIO-MediaCreation
+
       - name: Install dependencies
         run: |
+          . dist/bin/activate
           python -m pip install --upgrade pip
           pip install flake8 pytest pytest-cov OpenTimelineIO==0.13.0
-          pip install dependencies/OpenAssetIO
+          pip install dependencies/OpenAssetIO-MediaCreation
 
       - name: Install Plugin
         run: |
+          . dist/bin/activate
           pip install -e .
 
       - name: Lint with flake8
         run: |
+          . dist/bin/activate
           flake8 --show-source --statistics otio_openassetio tests
 
       - name: Test with pytest
         run: |
+          . dist/bin/activate
           pytest tests
         env:
           OPENASSETIO_PLUGIN_PATH: dependencies/OpenAssetIO/resources/examples/manager/BasicAssetLibrary/plugin
```

Comment:
[question] Not part of this PR, but the `bal_plugin_env` pytest fixture overwrites the env var, so is this here for a particular reason (similarly in the README instructions)?